meson.build

```diff
@@ -0,0 +1,220 @@
+
+package_version = meson.project_version()
+config_h.set_quoted(
+    'PACKAGE_VERSION', package_version,
+    description : 'Define to the version of this package.')
+
+package_string = '@0@ @1@'.format(package_name, package_version)
+config_h.set_quoted(
+    'PACKAGE_STRING', package_string,
+    description : 'Define to the full name and version of this package.')
+
+package_url = 'https://nixos.org/nix/'
+config_h.set_quoted(
+    'PACKAGE_URL', package_url,
+    description : 'Define to the home page for this package.')
+
+package_bug_url = 'https://github.com/nixos/nix/issues'
+config_h.set_quoted(
+    'PACKAGE_BUGREPORT', package_bug_url,
+    description : 'Define to the address where bug reports for this package should be sent.')
+
+
+# env
+#============================================================================
+
+
+# set install directories
+#-------------------------------------------------
+prefix = get_option('prefix')
+libdir = join_paths(prefix, get_option('libdir'))
+bindir = join_paths(prefix, get_option('bindir'))
+datadir = join_paths(prefix, get_option('datadir'))
+sysconfdir = join_paths(prefix, get_option('sysconfdir'))
+libexecdir = join_paths(prefix, get_option('libexecdir'))
+mandir = join_paths(prefix, get_option('mandir'))
+includedir = join_paths(prefix, get_option('includedir'))
+proj_inc = []
+
+# set nix directories
+#-------------------------------------------------
+
+# State should be stored in /nix/var, unless the user overrides it explicitly.
+if get_option('normal_var')
+    localstatedir = '/nix/var'
+else
+    localstatedir =  join_paths(prefix, get_option('localstatedir'))
+endif
+
+nixstoredir = get_option('nixstoredir')
+
+profiledir =  join_paths(sysconfdir, 'profile.d')
+
+
+# Construct a Nix system name (like "i686-linux").
+#-------------------------------------------------
+machine_name = host_machine.cpu()
+sys_name = host_machine.system().to_lower()
+
+cpu_archs = ['x86_64', 'armv6', 'armv7', '']
+
+foreach cpu : cpu_archs 
+    if host_machine.cpu().contains(cpu)
+        if cpu.contains('armv')
+            machine_name = cpu + '1'
+        else
+            machine_name = cpu
+        endif
+        break
+    endif
+endforeach
+
+system= '"' + machine_name + '-' + sys_name + '"'
+message('system name: ' + system)
+config_h.set(
+    'SYSTEM', system,
+    description : 'platform identifier (`cpu-os`)')
+
+
+# required dependencies
+#============================================================================
+
+
+# look for required programs
+#--------------------------------------------------
+cat = find_program('cat', required : true)
+ln = find_program('ln', required : true)
+cp = find_program('cp', required : true)
+rm = find_program('rm', required : true)
+bash = find_program('bash', required : true)
+echo = find_program('echo', required : true)
+patch = find_program('patch', required : true)
+xmllint = find_program('xmllint', required : true)
```

Comment:
I am not sure this is required...

`git grep xmllint` only shows commented lines (in scripts that seem to be specific to MacOS installs).

Reply:
could wrap it in a check for darwin if that makes more sense